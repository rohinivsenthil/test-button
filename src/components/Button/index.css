.round{
    border-radius: 50%;
    background-color: black;
    color: beige;
}

.rect{
    border: solid 1rem;
}