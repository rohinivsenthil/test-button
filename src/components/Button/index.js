import React, { useState} from 'react'
import * as styles from './index.css';

const Button = (props) => {

  const [count, setCount] = useState(0);

  const countHandler = () => {
    const {click}=props;
    setCount(count + 1)
    click();
  }

    const {text,buttonType,testId}=props;
    return (
      <button className={buttonType === 'rect' ? styles.rect : styles.round} data-testid={testId} onClick={() => countHandler()}>
        {text} clicked {count} times.
      </button> 
    )
  
}


export default Button;


