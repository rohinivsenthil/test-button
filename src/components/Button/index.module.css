.round{
    border-radius: 50%;
}

.rect{
    border: 1rem;
}