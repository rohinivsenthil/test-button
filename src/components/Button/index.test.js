import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import  Button  from './index';
import '../Button';


describe('the button component', () => {

  it('should render correctly', () => {
    const { asFragment } = render(<Button testID="test-btn">Test text</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call the onClick prop when the button is clicked', () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(<Button testId="test-btn" click={onClickMock}></Button>);

    fireEvent.click(getByTestId('test-btn'));
    expect(onClickMock).toHaveBeenCalled();
  });

  it('should render corrctly when the type:rect is passed to the component', () => {
    
    const { asFragment } = render(<Button testID="test-btn" buttonType="rect" />);

    expect(asFragment).toMatchSnapshot();
  });

  it('should render corrctly when the type:round is passed to the component', () => {
    const onClickMock = jest.fn();
    const { asFragment } = render(<Button testID="test-btn" buttonType="round" onClick={onClickMock} />);

    expect(asFragment).toMatchSnapshot();

  });

  it('should display the expected text on the button with clickCount',()=>{
    const onClickMock=jest.fn();
    const {getByTestId}=render(<Button testId="test-btn" buttonType="rect" click={onClickMock} text="Bhumika"/>);
    fireEvent.click(getByTestId("test-btn"));
    expect(getByTestId('test-btn')).toHaveTextContent("Bhumika clicked 1 times.");
  })
});
