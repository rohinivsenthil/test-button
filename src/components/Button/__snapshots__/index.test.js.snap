// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should render corrctly when the type:rect is passed to the component 1`] = `undefined`;

exports[`the button component should render corrctly when the type:round is passed to the component 1`] = `undefined`;

exports[`the button component should render correctly 1`] = `
<DocumentFragment>
  <button>
     clicked 0 times.
  </button>
</DocumentFragment>
`;
